test/bin/mirror_registry.sh

```diff
@@ -29,23 +29,19 @@ get_container_images() {
 prereqs() {
     mkdir -p "${REGISTRY_ROOT}"
     mkdir -p "${REGISTRY_CONTAINER_DIR}"
-    curl -L https://github.com/quay/mirror-registry/releases/download/v${QUAY_VERSION}/mirror-registry-offline.tar.gz -o /tmp/mirror-registry-offline.tar.gz
-    tar xf /tmp/mirror-registry-offline.tar.gz -C "${REGISTRY_ROOT}"
-    rm -f /tmp/mirror-registry-offline.tar.gz
     sudo dnf install -y podman skopeo jq
+    podman run -d -p 5000:5000 --restart always --name registry "quay.io/microshift/distribution:${DISTRIBUTION_VERSION}"
 }
 
 setup_registry() {
-    pushd "${REGISTRY_ROOT}" &>/dev/null
-    ./mirror-registry install -r "${REGISTRY_ROOT}" --initUser microshift --initPassword microshift
-    sudo cp "${REGISTRY_ROOT}/quay-rootCA/rootCA.pem" /etc/pki/ca-trust/source/anchors/mirror-registry.pem
-    sudo update-ca-trust
-    # The mirror does not allow anonymous access, therefore we need the user and password configured in the
-    # pull secrets.
-    podman login -u microshift -p microshift "${REGISTRY_HOST}" --authfile "${REGISTRY_ROOT}/local-auth.json"
-    jq -s '.[0] * .[1]' "${PULL_SECRET}" "${REGISTRY_ROOT}/local-auth.json" > "${PULL_SECRET}.tmp"
-    mv "${PULL_SECRET}.tmp" "${PULL_SECRET}"
-    popd &>/dev/null
+    # Docker distribution does not support TLS authentication. The mirror-images.sh helper uses skopeo without tls options
+    # and it defaults to https. Since this is not supported we need to configure registries.conf so that skopeo tries http instead.
+    sudo bash -c 'cat >> /etc/containers/registries.conf' << EOF
+[[registry]]
+location = "$(hostname)"
+insecure = true
+EOF
```

Comment:
This one also needs to be fixed not to append to the file when run a few times.
May I suggest just creating a file in `/etc/containers/registries.d`,  so it an be overwritten when run a few times.

Reply:
Done